assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.test.js

```diff
@@ -0,0 +1,303 @@
+	SITE_STATUS_ADDED,
+	SITE_STATUS_READY,
+} from '../../util';
+
+describe( 'AdSenseConnectCTA', () => {
+	let registry;
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus:
+			AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	};
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		provideSiteInfo( registry );
+		provideUserAuthentication( registry );
+		provideModules( registry, [
+			{
+				active: true,
+				connected: true,
+				slug: 'adsense',
+			},
+		] );
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'before click', () => {
+		it.each( [
+			[
+				'the Adsense module is not connected',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				false,
+				false,
+				1684145866,
+			],
+			[
+				'notification is dismissed',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				true,
+				1684145866,
+			],
+			[
+				'the Adsense account status is not ready',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the Adsense site status is not ready',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_ADDED,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the Ad blocking recovery status is an empty string',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the setup completed timestamp is less than three weeks',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+				true,
+				false,
+				1686836242,
+			],
+		] )(
+			'should not render the widget when %s',
+			(
+				testName,
+				accountStatus,
+				siteStatus,
+				adBlockingRecoverySetupStatus,
+				isModuleConnected,
+				isNotificationDismissed,
+				setupCompletedTimestamp
+			) => {
+				const Widget = ( { children } ) => <div>{ children }</div>;
+				const WidgetNull = () => <div>NULL</div>;
+
+				provideModules( registry, [
+					{
+						slug: 'adsense',
+						active: true,
+						connected: isModuleConnected,
+					},
+				] );
+
+				registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+					...validSettings,
+					accountStatus,
+					siteStatus,
+					adBlockingRecoverySetupStatus,
+					setupCompletedTimestamp,
+				} );
+				registry
+					.dispatch( CORE_USER )
+					.receiveGetDismissedItems( [
+						...( isNotificationDismissed
+							? [ 'ad-blocking-recovery-notification' ]
+							: [] ),
+					] );
+
+				const { container } = render(
+					<AdBlockingRecoveryWidget
+						Widget={ Widget }
+						WidgetNull={ WidgetNull }
+					/>,
+					{
+						registry,
+					}
+				);
+
+				expect( container ).toHaveTextContent( 'NULL' );
```

Comment:
Using a real `WidgetNull` will result in a change to this assertation:

```suggestion
				expect( container ).toBeEmptyDOMElement();
```

Reply:
Updated.